libnest/src/transaction/install.rs

```diff
@@ -73,64 +73,68 @@ impl InstallTransaction {
         let tarball_handle = npf_explorer
             .get_data()
             .with_context(|_| npf_path.display().to_string())?;
-        let mut tarball = tarball_handle.file();
-        let mut archive = Archive::new(GzDecoder::new(tarball));
-        let mut files = Vec::new();
-
-        // List all the files in the archive and check whether they already exist
-        for entry in archive
-            .entries()
-            .with_context(|_| npf_path.display().to_string())?
-        {
-            let entry = entry.with_context(|_| npf_path.display().to_string())?;
-            let entry_path = entry
-                .path()
-                .with_context(|_| npf_path.display().to_string())?;
-
-            let abs_path = Path::new("/").with_content(&entry_path);
-            let rel_path = config.paths().root().with_content(&entry_path);
-
-            // If the file exists and is not a directory, the installation would overwrite an
-            // existing file, return an error.
-            if let Ok(metadata) = fs::symlink_metadata(&rel_path) {
-                if !metadata.is_dir() {
-                    Err(format_err!("{}", abs_path.display())
-                        .context(InstallErrorKind::FileAlreadyExists))?;
+
+        if let Some(tarball_handle) = tarball_handle {
```

Comment:
Can we say that `Virtual` packages are now officially supported, and any failure in doing so is a bug and should be fixed?

Reply:
They are not at the moment, because we have yet to decide whether a log file should be created / read for those (and modify the remove transaction accordingly). Will PM you about this, ofc